app/jobs/scheduled/auto_expire_user_api_keys.rb

```diff
@@ -0,0 +1,15 @@
+module Jobs
+
+  class AutoExpireUserApiKeys < Jobs::Scheduled
+    every 1.day
+
+    def execute(args)
+      if SiteSetting.expire_user_api_keys_days > 0
+        expire_user_api_keys_days = SiteSetting.expire_user_api_keys_days.days.ago
+
+        UserApiKey.where("last_used_at < ?", expire_user_api_keys_days).update_all(revoked_at: Time.now)
```

Comment:
I think we need to think about clean up of revoked keys as well so we don't bloat the table over time.

Reply:
Do you think it would be fine to clean up all revoked keys older than `SiteSetting.delete_user_api_keys_days` (new site setting) or maybe something like `2 * SiteSetting.expire_user_api_keys_days`?